src/Serialization/Rings.jl

```diff
@@ -88,161 +95,185 @@ end
 
 ################################################################################
 # Multivariate and Universal Polynomials
-@registerSerializationType(QQMPolyRingElem)
-@registerSerializationType(ZZMPolyRingElem)
-@registerSerializationType(fqPolyRepMPolyRingElem)
-@registerSerializationType(zzModMPolyRingElem)
 @registerSerializationType(MPolyRingElem)
 @registerSerializationType(UniversalPolyRingElem)
 
 function save_internal(s::SerializerState, p::Union{UniversalPolyRingElem, MPolyRingElem})
     parent_ring = parent(p)
-    parent_ring = save_type_dispatch(s, parent_ring)
-    terms = []
+    base = base_ring(parent_ring)
+    parents = []
+    encoded_terms = []
 
     for i in 1:length(p)
-        term = Dict(
-            :exponent => save_type_dispatch(s, exponent_vector(p, i)),
-            :coeff => save_type_dispatch(s, coeff(p, i))
-        )
-        push!(terms, term)
+        if coeff == parent_ring(0)
+            continue
+        end
+        encoded_coeff = save_internal(s, coeff(p, i))
+        if has_elem_basic_encoding(base)
+            push!(encoded_terms,  (exponent_vector(p, i), encoded_coeff))
+        else
+            parents = encoded_coeff[:parents]
+            push!(encoded_terms, (exponent_vector(p, i), encoded_coeff[:terms]))
+        end
```

Comment:
Before the loop:
```julia
    parents = save_parents_internal(s, SOMETHING)
```
then here
```suggestion
        # no need to serialize the parents here, we already did;
        # in this case, the serialization of the coefficient does not
        # need to store the parents; and also it can directly store (return) the terms, instead of putting them into a sublist
        encoded_coeff = save_internal(s, coeff(p, i); omit_parents=true)
        push!(encoded_terms, (exponent_vector(p, i), encoded_coeff))
```

Reply:
resolved by adding an include_parents flag for save_internal of any ring elements, we use the get_parent_refs function now if the flag is true and returns a dict with a parents key, otherwise just returns the terms as a vector